File: MVVMSample/app/src/main/java/com/example/mvvmsample/view/MainActivity.java
package com.example.mvvmsample.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;
import android.util.Log;

import com.example.mvvmsample.R;
import com.example.mvvmsample.data.models.Todo;
import com.example.mvvmsample.viewmodel.TodoListViewModel;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ArrayList<Todo> todos = new ArrayList<>();
    TodoListViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewModel = ViewModelProviders.of(this).get(TodoListViewModel.class);
        viewModel.init();
        viewModel.getTodos().observe(this, response -> {
            todos.addAll(response);
            Log.e("Response", todos.size() + "");
        });
    }
}

File: MVPApplication/app/src/main/java/com/example/mvpapplication/ui/MainActivity.kt
package com.example.mvpapplication.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.mvpapplication.R
import com.example.mvpapplication.contract.ContractInterface
import com.example.mvpapplication.presenter.MainPresenter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), ContractInterface.View {

    private var presenter: MainPresenter? = null;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        presenter = MainPresenter(this)
    }

    override fun initView() {
        counterTextView.text = presenter?.getCounter()
        clickButton.setOnClickListener { presenter?.incrementValue() }
    }

    override fun updateViewData() {
        counterTextView.text = presenter?.getCounter()
    }
}
